projects/vdk-plugins/vdk-jupyter/vdk-jupyterlab-extension/src/__tests__/resquests.spec.ts

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright 2023-2023 VMware, Inc.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { requestAPI } from '../handler';
+import { VdkOption } from '../vdkOptions/vdk_options';
+import { getJobDataJsonObject, jobData, setJobDataToDefault } from '../jobData';
+import { jobdDataRequest, jobRequest, jobRunRequest } from '../serverRequests';
+
+jest.mock('../handler', () => {
+  return {
+    requestAPI: jest.fn()
+  };
+});
+
+window.alert = jest.fn();
+
+describe('jobdDataRequest', () => {
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('should call requestAPI with the correct arguments', async () => {
+    jobData.set(VdkOption.NAME, 'Test Job');
+    jobData.set(VdkOption.TEAM, 'Test Team');
+    jobData.set(VdkOption.PATH, '/test/job/path');
+
+    const mockData = {
+      [VdkOption.NAME]: 'Test Job',
+      [VdkOption.TEAM]: 'Test Team',
+      [VdkOption.PATH]: '/test/job/path'
+    };
+
+    (requestAPI as jest.Mock).mockResolvedValue(mockData);
+
+    await jobdDataRequest();
+
+    expect(requestAPI).toHaveBeenCalledTimes(1);
+    expect(requestAPI).toHaveBeenCalledWith('job', {
+      body: JSON.stringify(JSON.stringify(getJobDataJsonObject())),
```

Comment:
don't understand if the dynamic data is verified with the constant expected?

Reply:
jobData has the same values as the mockData so the mocked requestAPI should return json with the data of mockData, meanwhile getJobDataHsonObject returns the json with the data of jobData
So yes, the data is verified with the constant but indirectly